src/runDiff.ts

```diff
@@ -0,0 +1,225 @@
+  githubToken: string;
+  /** jobs.<job-id>.name of this workflow job. This is needed to get the job ID via query. */
+  jobName: string;
+  /** Name of the file this jobs outputs will be saved into. */
+  outputFilename: string;
+  /** The ref (branch or sha) to use with the diff. */
+  ref: string;
+  /** Full name of the CDKTF stack to diff. */
+  stack: string;
+  /** When present, no Terraform will execute. The output of this action will be substituted with the output contained in this file. This is useful for cases when you want to test but don't have authentication set up. */
+  stubOutputFile?: string;
+  /** The version of Terraform to use (defaults to 1.8.0) */
+  terraformVersion: string;
+  /** Working directory where CDKTF code is located */
+  workingDirectory: string;
+  /** Skip synthesis of the application, assume the synthesized Terraform code is already present and up to date */
+  skipSynth: boolean;
+  /** When given, attempt to download the given artifact contents into this working directory. */
+  artifactName?: string;
+}
+
+/**
+ * Interface representing the structured outputs from the action.
+ * These outputs can be used by subsequent steps in the workflow.
+ */
+interface ActionOutputs {
+  /** Direct link to the job execution */
+  html_url: string;
+  /** ID of the executed job */
+  job_id: string;
+  /** 
+   * Result code indicating the outcome:
+   * - '0': Success with no changes
+   * - '1': Error occurred
+   * - '2': Success with changes detected
+   */
+  result_code: "0" | "1" | "2";
+  /** Name of the CDKTF stack used */
+  stack: string;
+  /** Human-readable summary of changes */
+  summary: string;
+}
+
+/**
+ * Executes CDKTF diff command, parses the output, and provides structured results.
+ * Handles different output patterns to determine if there are changes, errors, or no changes.
+ */
+export class RunDiff {
+  octokit: Octokit & Api & { paginate: PaginateInterface };
+  context: Context;
+  inputs: ActionInputs;
+
+  /**
+   * Initialize clients and member variables.
+   */
+  constructor() {
+    // Get inputs first
+    this.inputs = {
+      githubToken: core.getInput("github_token", { required: true }),
+      jobName: core.getInput("job_name", { required: true }),
+      outputFilename: core.getInput("output_filename", { required: true }),
+      ref: core.getInput("ref", { required: true }),
+      stack: core.getInput("stack", { required: true }),
+      stubOutputFile: core.getInput("stub_output_file"),
+      terraformVersion: core.getInput("terraform_version") || "1.8.0",
+      workingDirectory: core.getInput("working_directory") || "./",
+      skipSynth: core.getBooleanInput("skip_synth"),
+      artifactName: core.getInput("artifact_name"),
+    };
+
+    // Then create octokit with the token
+    this.octokit = github.getOctokit(this.inputs.githubToken);
+    this.context = github.context;
+    core.debug("Context:");
+    core.debug(JSON.stringify(this.context));
```

Comment:
Just personal preference.
```suggestion
    core.debug(`Context:\n${JSON.stringify(this.context)}`);
```